### PR TITLE
move to mkdocs

### PR DESCRIPTION
**Pull request by dholbach**
First cut at moving docs over to `mkdocs` for flux too (follow up from https://github.com/fluxcd/helm-operator/pull/332)

### CODE REVIEW

**Comment by dholbach** [comment]
To review, run:

```console
pip3 install -r docs/requirements.txt
mkdocs serve
```

**Comment by dholbach** [comment]
The navbar on the left is something we can still tweak, and we might want to take a look at the currently quite empty index.md files.

**Comment by stefanprodan** [comment]
Can we have a preview link like you did for helm-op?

**Comment by dholbach** [comment]
Can do, it'll be manual though, as I don't think I can set up webhooks easily.

**Comment by dholbach** [comment]
https://test-flux.readthedocs.io/en/latest/

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- [Get Started](get-started/index.md)
+- [References](references/index.md)
+- [Guides](guides/index.md)
+- [Tutorials](tutorials/index.md)
+- [FAQ](faq.md)
+- [Troubleshooting](troubleshooting.md)
+
+## Helm Operator
+
+- Documentation <https://docs.fluxcd.io/projects/helm-operator/>
```
Comment (hiddeco) [inline docs/index.md]:
```suggestion
- [Documentation](https://docs.fluxcd.io/projects/helm-operator/)
```

**Comment by dholbach** [comment]
The landing page (docs/index.md) looks very boring right now as there's no "toctree: depth 2" kind of thing - maybe there's some plugin to do this? Maybe we should symlink the top-level README.md?

**Comment by stefanprodan** [comment]
@dholbach I think the index should contain the diagram and the feature list (like we did for helm-op). I'll make the feature list, please add the diagram for now.

**Comment by dholbach** [comment]
I think we're mostly good to go now.

**Comment by dholbach** [comment]
Current URLs for `latest` are:

```
https://docs.fluxcd.io/en/latest/
https://docs.fluxcd.io/en/latest/_static/css/theme.css
https://docs.fluxcd.io/en/latest/_static/pygments.css
https://docs.fluxcd.io/en/latest/_static/custom.css
https://docs.fluxcd.io/en/latest/genindex.html
https://docs.fluxcd.io/en/latest/search.html
https://docs.fluxcd.io/en/latest/introduction.html
https://docs.fluxcd.io/en/latest/requirements.html
https://docs.fluxcd.io/en/latest/get-started/index.html
https://docs.fluxcd.io/en/latest/references/index.html
https://docs.fluxcd.io/en/latest/guides/index.html
https://docs.fluxcd.io/en/latest/tutorials/index.html
https://docs.fluxcd.io/en/latest/faq.html
https://docs.fluxcd.io/en/latest/troubleshooting.html
https://docs.fluxcd.io/en/latest/contributing/index.html
https://docs.fluxcd.io/en/latest/references/blueprint.html
https://docs.fluxcd.io/en/latest/references/daemon.html
https://docs.fluxcd.io/en/latest/references/fluxctl.html
https://docs.fluxcd.io/en/latest/references/fluxyaml-config-files.html
https://docs.fluxcd.io/en/latest/references/garbagecollection.html
https://docs.fluxcd.io/en/latest/references/git-gpg.html
https://docs.fluxcd.io/en/latest/references/automated-image-update.html
https://docs.fluxcd.io/en/latest/references/helm-operator-integration.html
https://docs.fluxcd.io/en/latest/references/monitoring.html
https://docs.fluxcd.io/en/latest/guides/provide-own-ssh-key.html
https://docs.fluxcd.io/en/latest/guides/use-git-https.html
https://docs.fluxcd.io/en/latest/guides/use-private-git-host.html
https://docs.fluxcd.io/en/latest/guides/upgrading-to-1.0.html
https://docs.fluxcd.io/en/latest/tutorials/get-started.html
https://docs.fluxcd.io/en/latest/tutorials/get-started-helm.html
https://docs.fluxcd.io/en/latest/tutorials/get-started-kustomize.html
https://docs.fluxcd.io/en/latest/tutorials/driving-flux.html
https://docs.fluxcd.io/en/latest/contributing/get-started-developing.html
https://docs.fluxcd.io/en/latest/contributing/building.html
https://docs.fluxcd.io/en/latest/index.html
```

I'm wondering if we should add redirects for them all?

**Comment by hiddeco** [comment]
To prevent the projects vs organization mix-up:

```diff
diff --git a/mkdocs.yml b/mkdocs.yml
index fc3b535d..640d52a6 100644
--- a/mkdocs.yml
+++ b/mkdocs.yml
@@ -1,6 +1,6 @@
-site_name: Flux CD
+site_name: Flux
 # site_description:
-site_author: Flux CD Project
+site_author: Flux contributors
 site_url: https://docs.fluxcd.io/

 # Repository
```

`Roboto Mono` is already the default font for code, I also think a slightly smaller navigation title and logo look better:

```diff
diff --git a/docs/_static/custom.css b/docs/_static/custom.css
index fa1bff16..0c7edb7a 100644
--- a/docs/_static/custom.css
+++ b/docs/_static/custom.css
@@ -1,29 +1,24 @@
 @import url("https://fonts.googleapis.com/css?family=Montserrat&display=swap");
-@import url("https://fonts.googleapis.com/css?family=Roboto+Mono&display=swap");

 body {
   font-family: "Montserrat", sans-serif;
 }

-.rst-content pre.literal-block, .rst-content div[class^="highlight"] pre, .rst-content .linenodiv pre {
-  font-family: "Roboto Mono", monospace;
-}
-
-code {
-  font-family: "Roboto Mono", monospace;
-}
-
 .md-logo {
-    width: 54px;
-    height: 54px;
+    width: 40px;
+    height: 40px;
     padding-bottom: 2px;
     padding-top: 2px;
 }
 .md-logo img {
-    width: 44px;
-    height: 44px;
+    width: 40px;
+    height: 40px;
 }

-.md-header,.md-footer-nav {
+.md-header, .md-footer-nav {
     background-image: linear-gradient(45deg, rgb(0, 150, 225) 0%, rgb(27, 141, 226) 24%, rgb(42, 125, 227) 53%, rgb(53, 112, 227) 78%, rgb(53, 112, 227) 100%);
 }
+
+.md-header-nav__title {
+    font-size: .85rem;
+}
```

**Comment by dholbach** [comment]
nice one, thanks @hiddeco

**Comment by dholbach** [comment]
Apparently we can't use mkdocs-redirects for the above.
```
WARNING -  redirects plugin: 'introduction.html' is not a valid markdown file! 
```

**Comment by dholbach** [comment]
So yeah, for now this is all we can do. Please review and let me know when you want to land this. It'll require changing one setting from "sphinx" to "mkdocs" in the RTD UI.

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    #  - redirects:
+    #  redirect_maps:
+    #    'guides/upgrading-to-ga.md': 'how-to/upgrade-to-ga.md'
+    #    'guides/upgrading-to-beta.md': 'how-to/upgrade-to-beta.md'
+
+markdown_extensions:
+  - admonition
+  - codehilite:
+      guess_lang: false
+  - pymdownx.superfences
```
Comment (hiddeco) [inline mkdocs.yml]:
```suggestion
  - toc:
      permalink: true
  - pymdownx.superfences
```

Multi-line so not click-fixable.

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    - Get started: tutorials/get-started.md
+    - Get started using Helm: tutorials/get-started-helm.md
+    - How to bootstrap Flux using Kustomize: tutorials/get-started-kustomize.md
+    - Automations, locks and annotations: tutorials/driving-flux.md
+  - Frequently asked questions: faq.md
+  - Troubleshooting: troubleshooting.md
+  - Contributing:
+    - Get started developing: contributing/get-started-developing.md
+    - Building Flux: contributing/building.md
+
```
Comment (hiddeco) [inline mkdocs.yml]:
```suggestion
```

File: docs/references/fluxyaml-config-files.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Manifest generation through `.flux.yaml` configuration files
+# Manifest generation through .flux.yaml configuration files
```
Comment (hiddeco) [inline docs/references/fluxyaml-config-files.md]: Does using `` in the titles cause issues with something?

Comment (dholbach) [reply docs/references/fluxyaml-config-files.md]: It was broken when I first looked at things. Looks like this is fixed now - maybe with one of the mkdocs plugins we started using now? :man_shrugging: I'll revert the ` removals.

Comment (hiddeco) [reply docs/references/fluxyaml-config-files.md]: Mhmh, maybe the `pymdownx.superfences` plugin.

**Comment by dholbach** [comment]
Do we want something like this https://github.com/weaveworks/wksctl/pull/153/files#diff-b67911656ef5d18c4ae36cb6741b7965L150-R152? The `wksctl` team wanted it.

File: mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
+site_name: Flux
+# site_description:
+site_author: Flux Project
```
Comment (hiddeco) [inline mkdocs.yml]: Missed this last night, I changed this to  `The <project name> contributors` for the Helm operator as this seems to be inline with the ecosystem and CNCF projects.

File: mkdocs.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
+site_name: Flux
+# site_description:
```
Comment (hiddeco) [inline mkdocs.yml]:
```suggestion
site_description: Documentation for Flux, the GitOps Kubernetes operator.
```

(Creates a description tag for SEO purposes on the index page).

**Comment by hiddeco** [comment]
Awesome work, thanks a bunch @dholbach :star:

**Pull request merged**